docs/Internal language.md

```diff
@@ -0,0 +1,79 @@
+# The model language
+
+It is important to understand the domain-specific language that describes the data models. It is what the input processors converts their inputs to, and the generators use to generate the output.
+
+It is also what developers can use to create their own models, since they can provide a **raw data model** as input. This way you can create your own custom data models that can be interpreted to any output language, with the full sweep of features the generators and presets support. See [Create your own models from the ground up, instead of a supported input](./advanced#create-your-own-models-from-the-ground-up-instead-of-a-supported-input)
+
+## The basics and process
+
+There are two parts to the model, one for the input processors (the **raw data model**), which they convert to, and then the one that the generators receive (the **constrained data model**).
+
+For example (and this accounts for almost all languages) you cannot render a property with the name `my property`, generally, they follow some kind of common naming format such as using camel case `myProperty` or pascal case `MyProperty`.
+
+This is the reason for having two data models because each output (Java, TS, Go, etc) have very specific constraints.
+
+Therefore the **raw data model** does not have any constraints, and it is perfectly normal and expected to name your properties `my property`. Before the model reaches the generator, it gets transformed to a **constrained data model**.
+
+<p align="center">
+  <img src="./img/RenderingProcess.png" />
+</p>
+
+The transformation happens in three stages. 
+
+1. Process the input and transform it into the raw data model. See [The raw data model](#the-raw-data-model) for more information.
+2. Split the raw data model into separate models that are rendered separately. See [The splitting of data models](#The-splitting-of-data-models) for more information. 
+3. Constrain the data models to the output language. See [The constrained data model](#the-constrained-data-model) for more information.
+
+## The raw data model
```

Comment:
I think this should be move to the very top. You mentioned "raw data model" several times before this.

Reply:
Please see https://github.com/asyncapi/modelina/pull/530#issuecomment-1042000476